multiqc/utils/util_functions.py

```diff
@@ -73,12 +73,19 @@ def default(self, obj):
             try:
                 # Convert keys to strings
                 data = {str(k): v for k, v in data.items()}
-                # Get all headers
-                h = ["Sample"]
-                for sn in sorted(data.keys()):
-                    for k in data[sn].keys():
-                        if type(data[sn][k]) is not dict and k not in h:
-                            h.append(str(k))
+                # Get all headers from the data, except if data is a dictionary (i.e. has >1 dimensions)
+                # Use list -> set -> list to get only unique values
+                h = list(
+                    set(
```

Comment:
Are `set`s ordered? I don't think so, right? In which case we lose the original order of headers with this..

Reply:
> No, but they get sorted again a few lines below

That's only if the `sort_cols` config is set, which is usually isn't. Typically column order is preserved from the module.

Using dict keys is the way I would usually do this, sorry should have suggested that 👍🏻 Then they're unique and ordered (at least in Python 3, which MultiQC is now).